### PR TITLE
doc/user: polish release notes for v0.14.0

### PR DESCRIPTION
**Pull request by benesch**

### CODE REVIEW

File: doc/user/content/release-notes.md [hunk 1]
```diff
@@ -107,34 +107,26 @@ These changes are present in [unstable builds](/versions/#unstable-builds) and
...
-- **Breaking change.** Change the internal representation of numbers in
-  [`jsonb`](/sql/types/jsonb). Previously, JSON numbers were stored as either
-  [`int8`](/sql/types/int8) or [`float8`](/sql/types/float8) values; now they
-  are always stored as [`numeric`](/sql/types/numeric) values.
-
-- Support casts from [`timestamp`] and [`timestamp with time zone`] to
-  [`time`].
-
-- Add `pg_catalog.pg_roles` as a builtin view.
+- **Breaking change.** Disallow views with multiple unnamed columns.
```
Comment (morsapaes) [inline doc/user/content/release-notes.md:110]:
```suggestion
- **Breaking change.** Disallow views with multiple unnamed columns {{% gh 9413 %}}.
```

File: doc/user/content/release-notes.md [hunk 2]
```diff
@@ -155,24 +147,37 @@ changes that have not yet been documented.
...
-- Support the [`WITH ORDINALITY`] modifier for function calls in the `FROM`
+- **Breaking change.** When using an arbitrary expression in an `ORDER BY` or
+  `DISTINCT ON` clause, only recognize references to input columns. Previously,
+  Materialize would recognize references to output columns as well.
+
+  See the [column references](/sql/select#column-references) section of the
+  `SELECT` documentation for details.
+
+- Fix a bug that could cause wrong results in queries that used the `ROWS FROM`
+  clause. The bug occurred if functions beyond the second function in the clause
```
Comment (morsapaes) [inline doc/user/content/release-notes.md:158]:
```suggestion
- Fix a bug that could cause wrong results in queries that used the `ROWS FROM`
  clause {{% gh 9654 %}}. The bug occurred if functions beyond the second function in the clause
```

File: doc/user/content/release-notes.md [hunk 3]
```diff
@@ -107,34 +107,26 @@ These changes are present in [unstable builds](/versions/#unstable-builds) and
...
 
-- When using an arbitrary expression in an `ORDER BY` or `DISTINCT ON` clause,
-  only recognize references to input columns. Previously, Materialize would
-  recognize references to output columns as well.
+{{% version-header v0.14.0 %}}
 
-  See the [column references](/sql/select#column-references) section of the
-  `SELECT` documentation for details.
+- **Breaking change.** Change the internal representation of numbers in
+  [`jsonb`](/sql/types/jsonb). Previously, JSON numbers were stored as either
```
Comment (morsapaes) [inline doc/user/content/release-notes.md:127]:
```suggestion
- **Breaking change.** Change the internal representation of numbers in
  [`jsonb`](/sql/types/jsonb) {{% gh 9669 %}}. Previously, JSON numbers were stored as either
```

**Review by morsapaes — APPROVED** [review]
LGTM! For breaking changes, we should probably always provide a link to the relevant issue or fix — added those in.

**Comment by benesch** [comment]
> LGTM! For breaking changes, we should probably always provide a link to the relevant issue or fix — added those in.

Done!

File: doc/user/content/release-notes.md (hunk 2 above)
Comment (benesch) [reply doc/user/content/release-notes.md]: Just for posterity this one was actually an unreported bug, the fix for which was buried in #9686 (to which I've added a link). #9654 was a _different_ wrong results issue with `ROWS FROM` that shipped in the previous release.

**Pull request merged**